frontend/src/pages/FlightPage/components/FlightFilter/FlightFilter.tsx

```diff
@@ -0,0 +1,103 @@
+import './FlightFilter.css';
+import {AirlinesAsList, AirportsAsList, Filter} from "../../../../types/enum.ts";
+import {capitalizeFirstLetter} from "../../../../utils/funtioncs.ts";
+import {ChangeEvent, Dispatch, FormEvent, SetStateAction, useEffect, useState} from "react";
+
+type FlightFilterProps = {
+    selectedFilter: Filter,
+    setSelectedFilter: Dispatch<SetStateAction<Filter>>,
+}
+
+export default function FlightFilter({ selectedFilter, setSelectedFilter }: Readonly<FlightFilterProps>) {
+    const [filter, setFilter] = useState<Filter>({
+        airline: undefined,
+        origin: undefined,
+        destination: undefined
+    });
+    const [isDisabled, setIsDisabled] = useState<boolean>(true);
+
+    const handleChange = (event: ChangeEvent<HTMLSelectElement>) => {
+        const { name, value } = event.target;
+        setFilter({ ...filter, [name]: value === 'All' ? undefined : value })
```

Comment:
I think where you have used `<option>All</option>` the value is already automatically undefined, since you didn't set the value prop of the option, so you can just write value and i think it should work :)
```suggestion
        setFilter({ ...filter, [name]: value })
```